File: misc/aooswd/ex/bool_ex_interpreter/java/OrExp.java
/**
 * @author dgayler
 *
 */
public class OrExp implements BooleanExp
{

	private BooleanExp left;
	
	private BooleanExp right;
	
	public OrExp (BooleanExp left, BooleanExp right)
	{
		this.left = left;
		this.right = right;
	}
	
	/**
	 * @see BooleanExp#Evaluate(Context)
	 */
	public boolean evaluate(Context con)
	{
		return left.evaluate (con) || right.evaluate (con);
	}

	/**
	 * @see BooleanExp#generateCode(Context)
	 */
	public String generateCode(Context con)
	{
		String leftName = left.generateCode (con);
		String rightName = right.generateCode (con);
		con.addCodeStatement(new LoadStatement (leftName));
		con.addCodeStatement(new OrStatement (rightName));
		String temp = con.getTemp();
		con.addCodeStatement (new StoreStatement (temp));
		return temp;
	}
	
}

File: misc/aooswd/ex/bool_ex_interpreter/cpp/or_exp.cpp
#include "or_exp.h"
#include "load_statement.h"
#include "or_statement.h"
#include "store_statement.h"
#include "context.h"

string OrExp::generateCode (Context& c)
{
  string leftName = left -> generateCode (c);
  string rightName = right -> generateCode (c);
  c.addCodeStatement (new LoadStatement (leftName));
  c.addCodeStatement (new OrStatement (rightName));
  string temp = c.getTemp();
  c.addCodeStatement (new StoreStatement (temp));
  return temp;
}

